scripts/linters/pylint_extensions.py

```diff
@@ -611,6 +627,89 @@ def visit_functiondef(self, node):
         self.check_functiondef_yields(node, node_doc)
         self.check_docstring_style(node)
         self.check_docstring_section_indentation(node)
+        self.check_typeinfo(node, node_doc)
+
+    def check_typeinfo(self, node, node_doc):
+        """Checks whether all parameters in a function definition are
+        properly formatted.
+
+        Args:
+            node: astroid.node.Function. Node for a function or
+                method definition in the AST.
+            node_doc: Docstring. Pylint Docstring class instance representing
+                a node's docstring.
+
+        Raises:
+            Exc. Some exception.
+        """
+        # The regexes are taken from the pylint codebase and are modified
+        # according to our needs. Link: https://github.com/PyCQA/pylint/blob/
+        # e89c361668aeead9fd192d5289c186611ef779ca/pylint/extensions/
+        # _check_docs_utils.py#L428.
+        re_param_line = re.compile(
+            r"""
+            \s*  \*{{0,2}}(\w+)          # identifier potentially with asterisks
```

Comment:
Any specific reason for this formatting (sorry, a bit new to this)?

Reply:
This is taken from the pylint codebase, the formatting was same there, so I didn't changed the formatting, just modified it according to our use.